### PR TITLE
feat: replace precinct with esbuild

### PR DESCRIPTION
**Pull request by eduardoboucas**
**- Summary**

Replaces `precinct` with esbuild, purely for its parsing capabilities, in the legacy bundler flow.

Closes https://github.com/netlify/pod-serverless/issues/39.

**- Test plan**

N/A

**- A picture of a cute animal (not mandatory but encouraged)**

gophers-1924099_1920-1-1920x960

### CODE REVIEW

File: src/runtimes/node/list_imports.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const isBuiltinModule = require('is-builtin-module')
+const { tmpName } = require('tmp-promise')
+
+const { safeUnlink } = require('../../utils/fs')
+
+const getListImportsPlugin = ({ imports, path }) => ({
+  name: 'list-imports',
+  setup(build) {
+    build.onResolve({ filter: /.*/ }, (args) => {
+      const isImport = args.path !== path && !isBuiltinModule(args.path)
```
Comment (eduardoboucas) [inline src/runtimes/node/list_imports.js:11]: `precinct` has the `includeCore` property to exclude built-in Node modules, which doesn't exist in esbuild. So we use `is-builtin-module` to that instead.

File: src/runtimes/node/list_imports.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+      if (isImport) {
+        imports.add(args.path)
+      }
+
+      return {
+        namespace: 'list-imports',
+        external: isImport,
```
Comment (eduardoboucas) [inline src/runtimes/node/list_imports.js:19]: By setting a path as `external` we prevent the parser from traversing it.

File: src/node_dependencies/index.js [hunk 3]
```diff
@@ -43,36 +52,48 @@ const isNotJunk = function (file) {
...
   }
 
   state.localFiles.add(path)
 
   const basedir = dirname(path)
-  // This parses JavaScript in `path` to retrieve all the `require()` statements
-  // TODO: `precinct.paperwork()` uses `fs.readFileSync()` under the hood,
-  // but should use `fs.readFile()` instead
-  const dependencies = precinct.paperwork(path, { includeCore: false })
+  const dependencies = featureFlags.parseWithEsbuild
```
Comment (eduardoboucas) [inline src/node_dependencies/index.js:82]: We'll use this feature flag to roll this out progressively. Once the rollout is complete, we can remove `precinct` entirely.

**Review by netlify-team-account-1 — APPROVED** [review]

File: src/runtimes/node/list_imports.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  try {
+    await esbuild.build({
+      entryPoints: [path],
+      bundle: true,
+      outfile: targetPath,
+      platform: 'node',
+      plugins: [getListImportsPlugin({ imports, path })],
+    })
+  } finally {
+    safeUnlink(targetPath)
```
Comment (ehmicky) [inline src/runtimes/node/list_imports.js:45]:
```suggestion
    await safeUnlink(targetPath)
```

**Review by ehmicky — APPROVED** [review]

**Pull request merged**